pkgs/development/python-modules/fastbencode/default.nix

```diff
@@ -0,0 +1,22 @@
+{ lib, buildPythonPackage, fetchPypi, cython }:
+
+buildPythonPackage rec {
+  pname = "fastbencode";
+  version = "0.0.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-tryavlQtBmN5NSlXb0m6iJFQhVT4XQm11tXtevfgxuQ=";
+  };
+
+  nativeBuildInputs = [ cython ];
+
```

Comment:
```suggestion

  # has no tests
  doCheck = false;

```

Reply:
This package has tests:

```
$ nix-build -A python39Packages.fastbencode
...
test_decoder_type_error (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_decoder_type_error (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_dict (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_dict (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_dict_deepnested (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
...
test_string (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_string (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok

----------------------------------------------------------------------
Ran 56 tests in 0.006s

OK
Finished executing setuptoolsCheckPhase
/nix/store/mbd2n9anpch0i4xas9n5lciw560p69hq-python3.9-fastbencode-0.0.7
```
<details>
<summary>
full log
</summary>

```
$ nix-build -A python39Packages.fastbencode
this derivation will be built:
  /nix/store/g7svjvqj9cznf03avp8ccqwhpxv2wfcx-python3.9-fastbencode-0.0.7.drv
building '/nix/store/g7svjvqj9cznf03avp8ccqwhpxv2wfcx-python3.9-fastbencode-0.0.7.drv'...
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Using setuptoolsShellHook
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing setuptools-check-hook
Using setuptoolsCheckPhase
unpacking sources
unpacking source archive /nix/store/fxghsqcw1qrgfdxm7a2zkl4zyq7jrr4j-fastbencode-0.0.7.tar.gz
source root is fastbencode-0.0.7
setting SOURCE_DATE_EPOCH to timestamp 1629396402 of file fastbencode-0.0.7/setup.cfg
patching sources
configuring
no configure script, doing nothing
building
Executing setuptoolsBuildPhase
setup.py:23: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  cython_version_info = LooseVersion(cython_version)
setup.py:24: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  if cython_version_info < LooseVersion(minimum_cython_version):
running bdist_wheel
running build
running build_py
creating build
creating build/lib.macosx-10.12-x86_64-3.9
creating build/lib.macosx-10.12-x86_64-3.9/fastbencode
copying fastbencode/__init__.py -> build/lib.macosx-10.12-x86_64-3.9/fastbencode
copying fastbencode/_bencode_py.py -> build/lib.macosx-10.12-x86_64-3.9/fastbencode
running build_ext
skipping 'fastbencode/_bencode_pyx.c' Cython extension (up-to-date)
building 'fastbencode._bencode_pyx' extension
creating build/temp.macosx-10.12-x86_64-3.9
creating build/temp.macosx-10.12-x86_64-3.9/fastbencode
clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Ifastbencode -I/nix/store/4wqbim64yg4yr4a5wl50hf11ghhdqmz4-python3-3.9.12/include/python3.9 -c fastbencode/_bencode_pyx.c -o build/temp.macosx-10.12-x86_64-3.9/fastbencode/_bencode_pyx.o
clang-11: warning: argument unused during compilation: '-fno-strict-overflow' [-Wunused-command-line-argument]
clang -bundle -undefined dynamic_lookup -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-zlib-1.2.12/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-bzip2-1.0.6.0.2/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-expat-2.4.7/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xz-5.2.5/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libffi-3.4.2/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gdbm-1.23/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-sqlite-3.38.2/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-readline-6.3p08/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ncurses-6.3/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-openssl-1.1.1n/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-configd-453.19/lib -L/nix/store/vjgm931x9j65apyrb0hsv4cml7plm6cc-tzdata-2022a/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-zlib-1.2.12/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-bzip2-1.0.6.0.2/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-expat-2.4.7/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xz-5.2.5/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libffi-3.4.2/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gdbm-1.23/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-sqlite-3.38.2/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-readline-6.3p08/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ncurses-6.3/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-openssl-1.1.1n/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-configd-453.19/lib -L/nix/store/vjgm931x9j65apyrb0hsv4cml7plm6cc-tzdata-2022a/lib build/temp.macosx-10.12-x86_64-3.9/fastbencode/_bencode_pyx.o -L/nix/store/4wqbim64yg4yr4a5wl50hf11ghhdqmz4-python3-3.9.12/lib -o build/lib.macosx-10.12-x86_64-3.9/fastbencode/_bencode_pyx.cpython-39-darwin.so
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-zlib-1.2.12/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-bzip2-1.0.6.0.2/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-expat-2.4.7/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xz-5.2.5/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libffi-3.4.2/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gdbm-1.23/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-sqlite-3.38.2/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-readline-6.3p08/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ncurses-6.3/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-openssl-1.1.1n/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-configd-453.19/lib'
ld: warning: directory not found for option '-L/nix/store/vjgm931x9j65apyrb0hsv4cml7plm6cc-tzdata-2022a/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-zlib-1.2.12/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-bzip2-1.0.6.0.2/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-expat-2.4.7/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xz-5.2.5/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libffi-3.4.2/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gdbm-1.23/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-sqlite-3.38.2/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-readline-6.3p08/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ncurses-6.3/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-openssl-1.1.1n/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-configd-453.19/lib'
ld: warning: directory not found for option '-L/nix/store/vjgm931x9j65apyrb0hsv4cml7plm6cc-tzdata-2022a/lib'
/nix/store/1x6cnw18j9hx71mp231idqccfmyiqg35-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
installing to build/bdist.macosx-10.12-x86_64/wheel
running install
running install_lib
creating build/bdist.macosx-10.12-x86_64
creating build/bdist.macosx-10.12-x86_64/wheel
creating build/bdist.macosx-10.12-x86_64/wheel/fastbencode
copying build/lib.macosx-10.12-x86_64-3.9/fastbencode/_bencode_pyx.cpython-39-darwin.so -> build/bdist.macosx-10.12-x86_64/wheel/fastbencode
copying build/lib.macosx-10.12-x86_64-3.9/fastbencode/__init__.py -> build/bdist.macosx-10.12-x86_64/wheel/fastbencode
copying build/lib.macosx-10.12-x86_64-3.9/fastbencode/_bencode_py.py -> build/bdist.macosx-10.12-x86_64/wheel/fastbencode
running install_egg_info
running egg_info
writing fastbencode.egg-info/PKG-INFO
writing dependency_links to fastbencode.egg-info/dependency_links.txt
writing requirements to fastbencode.egg-info/requires.txt
writing top-level names to fastbencode.egg-info/top_level.txt
reading manifest file 'fastbencode.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching '*.pxd' under directory 'fastbencode'
writing manifest file 'fastbencode.egg-info/SOURCES.txt'
Copying fastbencode.egg-info to build/bdist.macosx-10.12-x86_64/wheel/fastbencode-0.0.7-py3.9.egg-info
running install_scripts
creating build/bdist.macosx-10.12-x86_64/wheel/fastbencode-0.0.7.dist-info/WHEEL
creating 'dist/fastbencode-0.0.7-cp39-cp39-macosx_10_12_x86_64.whl' and adding 'build/bdist.macosx-10.12-x86_64/wheel' to it
adding 'fastbencode/__init__.py'
adding 'fastbencode/_bencode_py.py'
adding 'fastbencode/_bencode_pyx.cpython-39-darwin.so'
adding 'fastbencode-0.0.7.dist-info/METADATA'
adding 'fastbencode-0.0.7.dist-info/WHEEL'
adding 'fastbencode-0.0.7.dist-info/top_level.txt'
adding 'fastbencode-0.0.7.dist-info/RECORD'
removing build/bdist.macosx-10.12-x86_64/wheel
Finished executing setuptoolsBuildPhase
installing
Executing pipInstallPhase
/private/tmp/nix-build-python3.9-fastbencode-0.0.7.drv-0/fastbencode-0.0.7/dist /private/tmp/nix-build-python3.9-fastbencode-0.0.7.drv-0/fastbencode-0.0.7
Processing ./fastbencode-0.0.7-cp39-cp39-macosx_10_12_x86_64.whl
Installing collected packages: fastbencode
Successfully installed fastbencode-0.0.7
/private/tmp/nix-build-python3.9-fastbencode-0.0.7.drv-0/fastbencode-0.0.7
Finished executing pipInstallPhase
post-installation fixup
strip is /nix/store/fjd0p8rk93db76l7wnk2jxh32x9yadfw-cctools-binutils-darwin-949.0.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/90sr8p5c105sg7bn3qq5395ns20ryxcs-python3.9-fastbencode-0.0.7/lib 
patching script interpreter paths in /nix/store/90sr8p5c105sg7bn3qq5395ns20ryxcs-python3.9-fastbencode-0.0.7
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
running install tests
no Makefile or custom installCheckPhase, doing nothing
pythonCatchConflictsPhase
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: fastbencode
setuptoolsCheckPhase
Executing setuptoolsCheckPhase
setup.py:23: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  cython_version_info = LooseVersion(cython_version)
setup.py:24: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  if cython_version_info < LooseVersion(minimum_cython_version):
running test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
running egg_info
writing fastbencode.egg-info/PKG-INFO
writing dependency_links to fastbencode.egg-info/dependency_links.txt
writing requirements to fastbencode.egg-info/requires.txt
writing top-level names to fastbencode.egg-info/top_level.txt
reading manifest file 'fastbencode.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching '*.pxd' under directory 'fastbencode'
writing manifest file 'fastbencode.egg-info/SOURCES.txt'
running build_ext
skipping 'fastbencode/_bencode_pyx.c' Cython extension (up-to-date)
building 'fastbencode._bencode_pyx' extension
clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Ifastbencode -I/nix/store/4wqbim64yg4yr4a5wl50hf11ghhdqmz4-python3-3.9.12/include/python3.9 -c fastbencode/_bencode_pyx.c -o build/temp.macosx-10.12-x86_64-3.9/fastbencode/_bencode_pyx.o
clang-11: warning: argument unused during compilation: '-fno-strict-overflow' [-Wunused-command-line-argument]
clang -bundle -undefined dynamic_lookup -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-zlib-1.2.12/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-bzip2-1.0.6.0.2/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-expat-2.4.7/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xz-5.2.5/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libffi-3.4.2/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gdbm-1.23/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-sqlite-3.38.2/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-readline-6.3p08/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ncurses-6.3/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-openssl-1.1.1n/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-configd-453.19/lib -L/nix/store/vjgm931x9j65apyrb0hsv4cml7plm6cc-tzdata-2022a/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-zlib-1.2.12/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-bzip2-1.0.6.0.2/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-expat-2.4.7/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xz-5.2.5/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libffi-3.4.2/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gdbm-1.23/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-sqlite-3.38.2/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-readline-6.3p08/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ncurses-6.3/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-openssl-1.1.1n/lib -L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-configd-453.19/lib -L/nix/store/vjgm931x9j65apyrb0hsv4cml7plm6cc-tzdata-2022a/lib build/temp.macosx-10.12-x86_64-3.9/fastbencode/_bencode_pyx.o -L/nix/store/4wqbim64yg4yr4a5wl50hf11ghhdqmz4-python3-3.9.12/lib -o /private/tmp/nix-build-python3.9-fastbencode-0.0.7.drv-0/fastbencode-0.0.7/fastbencode/_bencode_pyx.cpython-39-darwin.so
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-zlib-1.2.12/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-bzip2-1.0.6.0.2/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-expat-2.4.7/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xz-5.2.5/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libffi-3.4.2/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gdbm-1.23/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-sqlite-3.38.2/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-readline-6.3p08/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ncurses-6.3/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-openssl-1.1.1n/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-configd-453.19/lib'
ld: warning: directory not found for option '-L/nix/store/vjgm931x9j65apyrb0hsv4cml7plm6cc-tzdata-2022a/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-zlib-1.2.12/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-bzip2-1.0.6.0.2/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-expat-2.4.7/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-xz-5.2.5/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libffi-3.4.2/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gdbm-1.23/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-sqlite-3.38.2/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-readline-6.3p08/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ncurses-6.3/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-openssl-1.1.1n/lib'
ld: warning: directory not found for option '-L/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-configd-453.19/lib'
ld: warning: directory not found for option '-L/nix/store/vjgm931x9j65apyrb0hsv4cml7plm6cc-tzdata-2022a/lib'
test_decoder_type_error (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_decoder_type_error (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_dict (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_dict (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_dict_deepnested (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_dict_deepnested (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_empty_string (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_empty_string (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_int (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_int (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_junk (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_junk (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_large_string (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_large_string (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_list (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_list (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_list_deepnested (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_list_deepnested (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_long (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_long (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_malformed_dict (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_malformed_dict (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_malformed_int (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_malformed_int (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_malformed_list (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_malformed_list (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_malformed_string (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_malformed_string (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_string (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_string (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_unknown_object (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_unknown_object (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_unsupported_type (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_unsupported_type (fastbencode.tests.test_bencode.TestBencodeDecode) ... ok
test_bencached (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_bencached (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_bool (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_bool (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_dict (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_dict (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_dict_deep_nested (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_dict_deep_nested (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_int (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_int (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_invalid_dict (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_invalid_dict (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_list (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_list (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_list_as_tuple (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_list_as_tuple (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_list_deep_nested (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_list_deep_nested (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_long (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_long (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_string (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok
test_string (fastbencode.tests.test_bencode.TestBencodeEncode) ... ok

----------------------------------------------------------------------
Ran 56 tests in 0.006s

OK
Finished executing setuptoolsCheckPhase
/nix/store/mbd2n9anpch0i4xas9n5lciw560p69hq-python3.9-fastbencode-0.0.7
```
</details>

Reply:
``python setup.py test`` is deprecated. It would probably better to run the tests manually with ``${python.interpreter} -m unittest fastbencode.tests.test_suite``.